### PR TITLE
Ray GPU image with Torch 1.13 support

### PR DESCRIPTION
**Pull request by arnavgarg1**
This PR bumps the ludwig-ray and ludwig-ray-gpu images to Torch 1.13.1 and Cuda 11.6.
It also pins `psutil` to `5.9.4`

### CODE REVIEW

File: docker/ludwig-ray-gpu/Dockerfile [hunk 1]
```diff
@@ -33,14 +40,18 @@ RUN pip install -U pip
 WORKDIR /ludwig
 
 # TODO(travis): remove this once we fix CUDA 11.6 NCCL version in base image
-RUN pip install torch==1.12.1 torchtext torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu113
+RUN pip install torch==1.13 torchtext torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
```
Comment (tgaddair) [inline docker/ludwig-ray-gpu/Dockerfile:43]: Can remove `TODO` and use `torch==1.13.1`.

File: docker/ludwig-ray-gpu/Dockerfile [hunk 2]
```diff
@@ -33,14 +40,18 @@ RUN pip install -U pip
...
+
+# Downgrade NCCL to 2.11.4 to for compatibility with CUDA 11.6
+RUN sudo apt install --allow-change-held-packages --allow-downgrades -y \
```
Comment (tgaddair) [inline docker/ludwig-ray-gpu/Dockerfile:46]: Can you link to the Horovod issue here?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ on:
   schedule:
     - cron: "0 10 * * *" # everyday at 10am
   push:
-    branches: ["master", "release-*"]
+    branches: ["master", "release-*", "torch_113_image_upgrade"]
```
Comment (tgaddair) [inline .github/workflows/docker.yml:7]: Will want to remove this before landing, of course (just calling it out for completeness ;) ).

**Review by tgaddair — APPROVED** [review]
Looks great! Just a few nits.

File: .github/workflows/docker.yml (hunk 3 above)
Comment (arnavgarg1) [reply .github/workflows/docker.yml:7]: Haha absolutely! Will do one final round of testing with all of the updates and remove this :)

**Pull request merged**